wasmtime-api/src/wasm.rs

```diff
@@ -721,8 +888,449 @@ pub unsafe extern "C" fn wasm_trap_origin(_trap: *const wasm_trap_t) -> *mut was
 
 #[no_mangle]
 pub unsafe extern "C" fn wasm_trap_trace(_trap: *const wasm_trap_t, out: *mut wasm_frame_vec_t) {
-    let mut buffer = Vec::new();
-    (*out).size = 0;
-    (*out).data = buffer.as_mut_ptr();
-    mem::forget(buffer);
+    (*out).set_uninitialized(0);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn wasm_exporttype_name(et: *const wasm_exporttype_t) -> *const wasm_name_t {
+    if (*et).name_cache.is_none() {
+        let et = (et as *mut wasm_exporttype_t).as_mut().unwrap();
+        et.name_cache = Some(wasm_name_t::from_name(&et.ty.name()));
+    }
+    (*et).name_cache.as_ref().unwrap()
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn wasm_exporttype_type(
+    et: *const wasm_exporttype_t,
+) -> *const wasm_externtype_t {
+    let ty = Box::new(wasm_externtype_t {
```

Comment:
Sanity check (sorry I know this was an old, already merged PR): is this Box leaking?  From https://github.com/WebAssembly/wasm-c-api/blob/master/include/wasm.h#L281 it seems like the return value isn't marked as owned by the caller (plus it's `const`, which seems to usually imply not owned), so I want to ensure that the expectation here is that the caller should be calling `wasm_externtype_delete` on the returned value.

Reply:
It is leaking indeed. Can you file a separate issue for that? I'll review all similar constructs soon. The initial goal was to build Rust side of API, and we need review wasm-c-api.

Reply:
I'll file the issue.  I'm currently working on embedding Wasmtime in a .NET Core host so I happened to notice this while implementing the corresponding `wasm_importtype_*` functions which are needed so that I can bind module imports to host functions via reflection.

Reply:
Opened issue #343.